### ISSUE TITLE
JssProvider breaks injectSheet with enabled theming

### ISSUE THREAD

**Issue by HenriBeck** [cssinjs/react-jss#121]
Apparently, the theming isn't working with nested components that require theming.
It seems that the nested component will get the sheet and classes from the upper component.

I may have done something wrong but when I log some values, it seems that [this line](https://github.com/cssinjs/react-jss/blob/master/src/createHoc.js#L80) returns the sheet from the upper component which leads to not call the actual style creator function.

**Comment by kof** [comment]
I assume you did something wrong. Please provide code for reproduction.

**Comment by HenriBeck** [comment]
https://github.com/HenriBeck/materialize-react/tree/update-theming

Install packages and start storybook.
The upper component is the Background component.
The Ripple element should be working as the ripple does not depend on the theme.
The Button component, on the other hand, does not work and it will log out the classes for the Background component when rendered.

**Comment by kof** [comment]
> I can provide a more organized example by tomorrow.

That would be nice.

**Comment by HenriBeck** [comment]
https://github.com/HenriBeck/react-jss-theming-example

As you can see in the console, the button component does not log the correct object of classes and instead logs the classes of the Background component

**Comment by kof** [comment]
It sounds like you are passing classes down from your parrent and overriding JSS HOC classes by it.

**Comment by kof** [comment]
This sounds pretty much unrelated to theming.

**Comment by HenriBeck** [comment]
No, I'm not passing any classes down.
If I go into the react-jss package and log the [staticStylesheet](https://github.com/cssinjs/react-jss/blob/master/src/createHoc.js#L80) there, once it returns undefined for the Background component and for the button component it returns the sheet from the Background component. That's why I end up with different classes in the Button component.

**Comment by iamstarkov** [comment]
its odd, im investigating

**Comment by iamstarkov** [comment]
and yes, you are not passing classes from one component to another. its good

**Comment by iamstarkov** [comment]
you had some obvious errors, i fixed it in https://github.com/HenriBeck/react-jss-theming-example/pull/1

**Comment by iamstarkov** [comment]
TLDR: 
* you used composes wrong
* you used destructuring wrong in your styles creator functions

**Comment by iamstarkov** [comment]
and you used `JssProvider` when you didnt need

**Comment by iamstarkov** [comment]
nevertheless, problem is actually in JssProvider

**Comment by iamstarkov** [comment]
root problem is in JssProvider https://github.com/iamstarkov/react-jss-theming-example/pull/1

**Comment by iamstarkov** [comment]
![screen shot 2017-07-18 at 7 10 28 pm](https://user-images.githubusercontent.com/559321/28330145-cb03aafe-6bec-11e7-8006-6199d77a1c30.png)

right after you drop JssProvider everything goes nuts

![screen shot 2017-07-18 at 7 09 18 pm](https://user-images.githubusercontent.com/559321/28330131-bf745c42-6bec-11e7-88a7-fd173a6a6dbe.png)

**Comment by HenriBeck** [comment]
Hmmm.... So I can't use JssProvider with ThemeProvider

**Comment by iamstarkov** [comment]
you should be able, once we fixed it

**Comment by HenriBeck** [comment]
No Problem

**Comment by iamstarkov** [comment]
it has nothing to do with theming

**Comment by kof** [comment]
We have no tests using JssProvider together with ThemeProvider.

**Comment by iamstarkov** [comment]
@kof disregard wrong usage — i was wrong there

**Comment by iamstarkov** [comment]
failing test https://github.com/cssinjs/react-jss/pull/122

**Comment by kof** [comment]
We made a very odd mistake, fixing it.

**Comment by kof** [comment]
Got a fix https://github.com/cssinjs/react-jss/pull/122 will be released today after CR.

**Comment by HenriBeck** [comment]
I'm glad this has been fixed so fast👍

**Comment by iamstarkov** [comment]
released as `7.0.2`
